native/src/main/java/io/ballerina/stdlib/grpc/ServerCall.java

```diff
@@ -82,20 +79,16 @@ public void sendHeaders(HttpHeaders headers) {
             throw new IllegalStateException("call is closed");
         }
         outboundMessage.removeHeader(GrpcConstants.MESSAGE_ENCODING);
-        if (compressor == null) {
-            compressor = Codec.Identity.NONE;
-        } else {
-            if (messageAcceptEncoding != null) {
-                List<String> acceptEncodings = Arrays.stream(messageAcceptEncoding.split("\\s*,\\s*"))
-                        .map(mediaType -> mediaType.split("\\s*;\\s*")[0])
-                        .collect(Collectors.toList());
-                if (!acceptEncodings.contains(compressor.getMessageEncoding())) {
-                    // resort to using no compression.
-                    compressor = Codec.Identity.NONE;
-                }
+
+        if (headers != null && headers.contains(GrpcConstants.MESSAGE_ENCODING)) {
+            String messageEncodingType = headers.get(GrpcConstants.MESSAGE_ENCODING);
+            if (compressorRegistry.lookupCompressor(messageEncodingType) != null) {
```

Comment:
```suggestion
            compressor = compressorRegistry.lookupCompressor(messageEncodingType);
            if (compressor == null) {
                compressor = Codec.Identity.NONE;
            }
```
Can we write it like this?

Reply:
Added it in here https://github.com/ballerina-platform/module-ballerina-grpc/pull/435/commits/3529e48c03f9df5acbf60c735abe9c2f2d5adfb3